starlite/kwargs.py

```diff
@@ -307,16 +308,14 @@ async def get_request_data(self, request: Request) -> Any:
             return parse_form_data(media_type=media_type, form_data=form_data, field=model_field)
         return await request.json()
 
-    async def resolve_dependency(
-        self, dependency: Dependency, connection: Union[WebSocket, Request], **kwargs: Any
-    ) -> Any:
+    async def resolve_dependency(self, dependency: Dependency, method: str, url: URL, **kwargs: Any) -> Any:
```

Comment:
Can you talk about why it is better to pass the method and url around instead of the connection? I have not problems with it but just curious.

Reply:
- It is an abstraction. Soon, I will add unit tests that will test every little detail and case. If the function takes both `Request` or `Websocket`, I’d have to double all tests against this method.
- Separation of concerns: why `KwargsModel` has to know about connection classes?
- Separation of concerns, again.  When we pass `Request` or `Websocket` to `KwargsModel`, it does nothing with it but extract necessary data. If we give the method just simple _data_ values—method and URL—mental load goes down when you read the code.

Reply:
>  I’d have to double all tests against this method.

Something easily done with pytest parametrization.

I see all your points and we are getting nitty gritty here, so no wrong answers. I see two parameters to every function instead of one, and lost context (i.e., there's nothing we cannot do with the existing `connection: ...` arg, but plenty we cannot do without it - but hypothetical of course. 

We are in the weeds here really, so I don't care overall.. but I think it borders on changes for the sake of changes. If this was new code I wouldn't bat an eyelid.

Reply:
> Something easily done with pytest parametrization.

Of course it is. I don't debate that it is. It just can feel like a burden to maintain and adds cognitive load.

I reverted this change and added `SignatureModel.get_connection_method_and_url` static method. Now I'm not sure it belongs in `SignatureModel` 🤔

Reply:
> Now I'm not sure it belongs in SignatureModel

Haha, well maybe not - make the call if you think it needs to move - but don't let perfect be the enemy of good enough:)

For the record, I was going to be happy if you left it, or changed it back - just good to talk about these things.

Good stuff.

Reply:
> For the record, I was going to be happy if you left it, or changed it back - just good to talk about these things.

I understand. Changed my mind—that's all. The goal of this PR is to reduce code complexity, not increase it 😂